activerecord/lib/active_record/connection_adapters/postgresql/oid/type_map_initializer.rb

```diff
@@ -28,8 +28,8 @@ def run(records)
             mapped.each     { |row| register_mapped_type(row)    }
             enums.each      { |row| register_enum_type(row)      }
             domains.each    { |row| register_domain_type(row)    }
-            arrays.each     { |row| register_array_type(row)     }
             ranges.each     { |row| register_range_type(row)     }
+            arrays.each     { |row| register_array_type(row)     }
```

Comment:
Some `arrays` is based on `ranges` so that if runing `register_array_type(row)` before `register_range_type(row)`, some array types may be skipped while invoking https://github.com/rails/rails/blob/beed933b9bb9f317697a95a36da067a67797e0d8/activerecord/lib/active_record/connection_adapters/postgresql/oid/type_map_initializer.rb#L109-L115